File: Examples/ObjC/FullUI/SlyceSDK-FullUI-Demo-ObjC/SearchDetail/SearchDetailViewController.m

#import "SearchDetailViewController.h"

@interface SearchDetailViewController ()

@end

@implementation SearchDetailViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self updateItemTextView];
}

- (IBAction)close:(UIButton *)sender {
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void)setItem:(NSDictionary *)item {
    _item = item;
    [self updateItemTextView];
}

- (void)updateItemTextView {
    self.itemTextView.text = [self.item description];
}

@end

File: Examples/Swift/FullUI/SlyceSDK-FullUI-Demo-Swift/SearchDetail/SearchDetailViewController.swift

import UIKit

class SearchDetailViewController: UIViewController {
    
    @IBOutlet weak var itemTextView: UITextView?
    
    var item: [AnyHashable: Any]? {
        didSet {
            updateItemTextView()
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        updateItemTextView()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func close(_ sender: UIButton) {
        self.dismiss(animated: true, completion: nil)
    }
    
    fileprivate func updateItemTextView() {
        self.itemTextView?.text = self.item?.description
    }

}
